### DISCUSSION TITLE
Combo to "untap" tap command [ZMK, 34-key a.dux]

### DISCUSSION THREAD

**Discussion by miatriestogrok** [manna-harbour/miryoku#228] [Q&A] [answered]
Dear all,

as a complete beginner, I am learning to touch type with an a.dux (ZMK). Because I still type slowly, I inadvertently activate the home row mods. I was looking for a way to exclude them for a certain time (until I need alt+gr or similar for european languages). Out of the following options, what would be the easiest to implement and how do I go about it:

1. do an "untap" custom key binding as opposite of "tap" on additional feature layer (e.g.  one or two key presses of lower fourth finger on additional feature layer). 
Manual states: "Returning from Tap requires a reset or adding a custom keybinding". Even though a key binding is a more elegant option, I am not quite sure what a "reset" would mean - if it is flashing, it would be a bit cumbersome in my workflow. However, I am not sure whether custom combos are supported yet.

2. add "tap" layer as an extra layer: limited by 34-key setup - all thumb keys in use (although I could merge mouse and media layer to free one up, but as above - total beginner and might not be the easiest choice?)

3. increase time before home row mods activate - from what I can see, on the issue list but not implemented at the moment.

Thank you very much for your kind support!

**Comment by manna-harbour** [comment, marked as answer]
Typing slowly is fine, but try to type each keystroke with light snappy taps without lingering on the press.

1. Using the Tap layer is ok for practicing the alphas but not for general use as you lose access to layers. Reset is a power cycle or press of hardware reset button. You can customise Tap or the mapping to add a key to return to Base using https://github.com/manna-harbour/miryoku/discussions/85. Example keycodes to use can be seen here https://github.com/manna-harbour/miryoku_zmk/blob/12fbc08bc2e8f257b13bd23168fdb558f0ebd1f1/miryoku/miryoku_babel/miryoku_layer_alternatives.h#L276.

2. Switching to Tap is already supported, but see 1. on switching back to Base.

3. Edit https://github.com/manna-harbour/miryoku_zmk/blob/12fbc08bc2e8f257b13bd23168fdb558f0ebd1f1/miryoku/miryoku.h#L27

4. Another approach would be to remove mod tap from Extra but keep layer tap, and mods on sub-layers, using the customisation mentioned mentioned above.

5. Alternatively remove mod tap from Base and Extra by replacing https://github.com/manna-harbour/miryoku_zmk/blob/12fbc08bc2e8f257b13bd23168fdb558f0ebd1f1/miryoku/miryoku_behaviors.h#L6 with 
```
#define U_MT(MOD, TAP) TAP
```
